src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/ApplicationModelConventionExtensions.cs

```diff
@@ -167,6 +187,36 @@ public void Apply(ApplicationModel application)
             }
         }
 
+        private class ParameterBaseApplicationModelConvention :
```

Comment:
We could use this patterns in all the other conventions that we implement in this class, but it's a breaking behavior change - where `Apply(applicationModel)` would have worked in the past, you now have to ensure it's executed via convention executor to get it to apply to the controller \ actions \ properties. Removes a bunch of code duplication, so might be worth considering.

Reply:
We don't want the breaking changes.

I'm not worried about code duplication in this case, this code is all structure, it doesn't have an signficant logic and is unlikely to change.